src/libYARP_sig/src/yarp/sig/Vector.h

```diff
@@ -119,55 +120,30 @@ template<class T>
 class yarp::sig::VectorOf : public VectorBase
 {
 private:
-    yarp::os::ManagedBytes bytes;
-    T *first;
-    size_t len;
-
-    inline void _updatePointers() {
-        len = bytes.used()/sizeof(T);
-        if (len==0) {
-            first = nullptr;
-        } else {
-            first = reinterpret_cast<T*>(bytes.get());
-        }
-    }
+    std::vector<T> bytes;
 
 public:
-    typedef T* iterator;
-    typedef const T* const_iterator;
 
     VectorOf() {
-        bytes.allocate(16*sizeof(T)); // preallocate space for 16 elements
-        bytes.setUsed(0);
-        first = nullptr;
-        len = 0;
+        bytes.reserve(16); // preallocate space for 16 elements
```

Comment:
Do we still need this? I don't think we need to keep compatibility with the old behaviour

Reply:
In the past I saw people using empty `Vectors` to retrieve data from our interfaces, that it works because our vector it is initialize with 16 elements by defauit(max n of joints I suppose).

And I saw crashes when passing to std::vector. I know that it is a very bad practice, but changing the behaviour could make crash (bad)software here and there.

Reply:
I think that's a sick behaviour, and we should not be allowed.

Reply:
If the interface takes a Vector as input then it's its responsibility to allocate the necessary memory. Thus, I concur with @drdanz to remove the preallocation.

Moreover, if you declare
```c++
yarp::sig::Vector v;
```
`v` is assumed to be empty and e.g. `v[0]` will generate a run-time error already in the current behavior and that's ok.

Reply:
I put `yarp::sig::Vector() = default` then.